src/query/storage/m3/storage.go

```diff
@@ -105,13 +105,20 @@ func (s *m3storage) FetchBlocks(
 	query *storage.FetchQuery,
 	options *storage.FetchOptions,
 ) (block.Result, error) {
-	if options.UseLegacy {
+	opts := s.opts
+	switch options.BlockType {
```

Comment:
this flow is a little confusing since in only one case you return and others you don't

Reply:
one way to avoid that could be to move everything after the switch into another function and then call that both from models.TypeMultiBlock and also a default case

Reply:
Switched it to `if` to make it a bit clearer, you're right that there's not a lot of point on a `switch` there